src/Telemetry/Microsoft.DotNet.AzureDevOpsTimelineTests/AzureDevOpsTimelineTimeoutTests.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            public static void Dependencies(IServiceCollection collection)
+            {
+
+                collection.AddOptions();
+                collection.AddLogging(logging =>
+                {
+                    logging.AddProvider(new NUnitLogger());
+                });
+                collection.Configure<AzureDevOpsTimelineOptions>((o, p) =>
+                {
+                    o.LogScrapingTimeout = "00:00:100";
+                });
+            }
+
+            public static Func<IServiceProvider, AzureDevOpsTimeline> Controller(IServiceCollection collection)
+            {
+                collection.AddScoped<AzureDevOpsTimeline>();
+                return s => s.GetRequiredService<AzureDevOpsTimeline>();
+            }
+
+            [ConfigureAllParameters]
+            public static Func<IServiceProvider, InMemoryTimelineTelemetryRepository> Repository(
+                IServiceCollection collection, string project, DateTimeOffset latestTime)
+            {
+                collection.AddSingleton<ITimelineTelemetryRepository>(
+                    s => new InMemoryTimelineTelemetryRepository(
+                        new List<(string project, DateTimeOffset latestTime)> { (project, latestTime) }
+                    )
+                );
+
+                return s => (InMemoryTimelineTelemetryRepository)s.GetRequiredService<ITimelineTelemetryRepository>();
+            }
+
+            public static void Clock(IServiceCollection collection, DateTimeOffset staticClock)
+            {
+                Mock<ISystemClock> mockSystemClock = new Mock<ISystemClock>();
+                mockSystemClock.Setup(x => x.UtcNow).Returns(staticClock);
+                collection.AddSingleton(mockSystemClock.Object);
+            }
+
+            public static void Build(IServiceCollection collection, BuildAndTimeline build, HttpMessageHandler httpMessageHandler)
+            {
+                collection.AddSingleton<IAzureDevOpsClient>(client => new MockTimeoutAzureClient(new Dictionary<Build, List<Timeline>>
+                {
+                    {build.Build, build.Timelines.ToList()}
+                }, httpMessageHandler));
+                collection.AddSingleton<IBuildLogScraper, BuildLogScraper>();
+            }
+        }
+
+        [Test]
+        public async Task TestAzureDevOpsTimelineTimeout()
+        {
+            DateTimeOffset timeDatum = DateTimeOffset.Parse("2021-01-01T01:00:00Z");
+            string azdoProjectName = "public";
+            string targetBranchName = "theTargetBranch";
+            MockHttpMessageHandler httpMessageHandler = MockHttpMessageHandler.Create("some string");
+
+            BuildAndTimeline build = BuildAndTimelineBuilder.NewPullRequestBuild(1, azdoProjectName, targetBranchName)
+                .AddTimeline(TimelineBuilder.EmptyTimeline("1", timeDatum)
+                    .AddRecord("NetCore1ESPool-Internal 5", "Initialize job", "https://www.microsoft.com"))
```

Comment:
It would be best to use a url that ended in ".test", so that we don't accidentally make real web requests out of here. "dev.azure.test" should work.  ".test" is a reserved top level domain that will never resolve to anything, so no real request can be sent.